## scipy.optimize

In [4]:
from scipy.optimize import minimize
def objective_function(x):
    return x**2 + 3*x + 2
result = minimize(objective_function, x0 = 0)
print(result.x)

[-1.50000001]


## scipy.integrate

In [52]:
from scipy.integrate import quad
def integrand(x):
    return x**2

result, error = quad(integrand, 0 , 1)
print(f"{error:.2f}")



0.00


- Error: This is an estimate of how far the computed result might be from the exact mathematical solution of the integral.


- here 0 , 1 are upper limit and lower limmit

## scipy.linalg

In [40]:
from scipy.linalg import inv,det
import numpy as np
matrix=np.array([[1,2],[3,4]])
matrix_det=det(matrix)
print(matrix_det)

-2.0


In [46]:
x = np.array([[4,7],[2,6]])
x_inv = inv(x)
print(x_inv)

[[ 0.6 -0.7]
 [-0.2  0.4]]


## scipy.interpolate

In [32]:
from scipy.interpolate import interp1d
x=np.array([0,1,2,3])
y=np.array([0,1,4,9])
# imterpolation function
f=interp1d(x,y,kind='linear')
# interpolating a new value
new_x=2.5
interpolated_value=f(new_x)
print(interpolated_value)

6.5


- basixally for 2.5 value in x array the value in y will be 6.5